rio_tiler/utils.py

```diff
@@ -244,9 +244,38 @@ def get_vrt_transform(
 
     """
     if src_dst.crs != dst_crs:
+        src_width = src_dst.width
+        src_height = src_dst.height
+        src_bounds = list(src_dst.bounds)
+
+        # Fix for https://github.com/cogeotiff/rio-tiler/issues/654
+        #
+        # When using `calculate_default_transform` with dataset
+        # which span at high/low latitude outside the area_of_use
+        # of the WebMercator projection, we `crop` the dataset
+        # to get the transform (resolution).
+        #
+        # Note: Should be handled in gdal 3.8 directly
+        # https://github.com/OSGeo/gdal/pull/8775
+        if (
+            src_dst.crs == WGS84_CRS
+            and dst_crs == WEB_MERCATOR_CRS
+            and (src_bounds[1] < -85.06 or src_bounds[3] > 85.06)
+        ):
```

Comment:
In theory we should do this for all projection combination but rasterio does not surface CRS `area of use`